architecture/design/ysql-gin-indexes.md

```diff
@@ -0,0 +1,107 @@
+# YSQL GIN indexes
+
+As of v2.6/v2.7, Yugabyte supports only one type of index access method for DocDB-backed relations: `lsm`.  This simply maps columns to the primary key of the base table.
+
+Generalized inverted indexes map elements inside container columns to the pk of the base table.  To support GIN, we add a new access method `ybgin` and implement the access method api.  A lot of the work can be borrowed from upstream postgres's `gin` access method.
+
+## opclasses
+
+Upstream postgres comes with the following [`gin` opclasses][opclasses]:
+
+- `tsvector_ops`: map `tsvector` to `text`
+- `array_ops`: map `anyarray` to `anyelement`
+- `jsonb_ops`: map `jsonb` to `text` (key or value in jsonb)
+- `jsonb_path_ops`: map `jsonb` to `int4` (hash of root-to-leaf path in jsonb)
+
+We can just duplicate these for the `ybgin` access method and reference the same opfamily.
+
+[opclasses]: https://www.postgresql.org/docs/current/gin-builtin-opclasses.html
+
+## amapi
+
+### build
+
+`ambuild` and `yb_ambackfill` are used for `CREATE INDEX` with and without online schema changes.  They both read from the base table and insert to the index table.  Therefore, this is a superset of `yb_aminsert`.  There's not much else to it.
```

Comment:
"There's not much else to it." should be deleted.

Reply:
It conveys additional information that, although it's a superset, it doesn't contain much else that should be noted here.